jac-uk/apply
# #798 Apply for a staged application

## What's included?
Enable candidates to apply for a vacancy and provide additional information when requested during the application process
See corresponding functionality in Admin https://github.com/jac-uk/admin/pull/1362

## Who should test?
✅ Developers
✅ Product owner
✅ User testing Group

## How to test?
Below are tests for the expected behavior of Apply.
There are corresponding test for Admin - see admin#1362

### 1. Apply for a vacancy (which has a staged application process)
 - In Admin configure an exercise to have a staged application process
 - Follow the vacancy listing and advert links to apply for the vacancy
 - On the **Apply for the role** (task list) page notice that only the configured information is being requested
 - Notice that the Review button at the bottom of the page is disabled
 - Follow the links to fill in the requested pieces of information noticing that when each form is saved  'DONE' appears next to each link in the task list
 - Notice that when all pieces of information have been filled in - i.e. each link has 'DONE' next to it - the Review button at the bottom of the page is active
 - Click the Review button to get to the **Review** page
 - Notice that the Review page shows only the information requested
 - Notice that each piece of information has a 'Change' link next to it to enable changes to be made
 - Check that changes can be made
 - Click the 'Send application' button to get to the **Application sent** (confirmation) page
 - Notice that the confirmation page includes the application reference number along with a message that indicates there are further stages to the application process
 - Click the 'dashboard' link to view all current applications
 - Notice that the vacancy applied for appears in the list of applications

**Expected:** the above steps can be carried out successfully

**Video:** https://drive.google.com/file/d/1a2tmpTLM7prT-gXYh_76aorJBuU4CwNp/view?usp=sharing


### 2. Continue applying for a vacancy (which has a staged application process)
 - In Admin progress an exercise to the next stage of the application process AND ensure applications are in the desired stage/status (e.g. Passed Tests)
 - Log in to a candidate account and navigate to the **Applications** list
 - (when the stage is open AND the application has the correct status) Notice the application has a 'Continue with application' button
 - Check that the Continue button only shows up when it should by changing the dates/stage in Admin
 - Click the Continue button to get to the **Apply for the role** (task list) page
 - Notice that information requested during registration is indicated however there are no links to make changes
 - Notice that the correct new information is being requested and that there are links to provide the information
 - Notice that the Review button at the bottom of the page is disabled
 - Follow the links to fill in the requested pieces of information noticing that when each form is saved  'DONE' appears next to each link in the task list
 - Notice that when all pieces of information have been filled in - i.e. each link has 'DONE' next to it - the Review button at the bottom of the page is active
 - Click the Review button to get to the **Review** page
 - Notice that the Review page shows the information requested at this stage of the process as well as information requested earlier in the process (e.g. at registration)
 - Notice that each piece of information requested at this stage has a 'Change' link next to it to enable changes to be made
 - Check that changes can be made
 - Notice that each piece of information requested previously (e.g. during registration) cannot be changed
 - Click the 'Send application' button to get to the **Application sent** (confirmation) page
 - Notice that the confirmation page includes the application reference number along with a message that indicates there are further stages to the application process
 - Click the 'dashboard' link to view all current applications
 - Notice that the application appears in the list of applications and is still editable
 - Test that the application is no longer editable when the current application stage closes

**Expected:** the above steps can be carried out successfully

**Video:** https://drive.google.com/file/d/1unI5x9HpKgf2WMH9f3mDh9v8FH5vVQUy/view?usp=sharing

## Risk - how likely is this to impact other areas?
🟠 Medium risk - this does change code that is shared with other areas

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
